example/glue/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -223,10 +224,73 @@ MM_CollectorLanguageInterfaceImpl::scavenger_internalGarbageCollect_shouldPercol
 }
 
 GC_ObjectScanner *
-MM_CollectorLanguageInterfaceImpl::parallelScavenger_getObjectScanner(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr, void *allocSpace, uintptr_t flags)
+MM_CollectorLanguageInterfaceImpl::scavenger_getObjectScanner(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr, void *allocSpace, uintptr_t flags)
 {
-#error provide an object scanner
+#if defined(OMR_GC_MODRON_SCAVENGER_STRICT)
+	Assert_MM_true((GC_ObjectScanner::scanHeap == flags) ^ (GC_ObjectScanner::scanRoots == flags));
+#endif /* defined(OMR_GC_MODRON_SCAVENGER_STRICT) */
+	GC_ObjectScanner *objectScanner = NULL;
+	objectScanner = GC_MixedObjectScanner::newInstance(env, objectPtr, allocSpace, flags);
+	return objectScanner;
 }
+
+void
+MM_CollectorLanguageInterfaceImpl::scavenger_flushReferenceObjects(MM_EnvironmentStandard *env)
+{
+	/* Do nothing for now */
+}
+
+bool
+MM_CollectorLanguageInterfaceImpl::scavenger_hasIndirectReferentsInNewSpace(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr)
+{
```

Comment:
The assert has to come before the return but I am not sure it should be an assert.  the contract right now for the glue layer is that we use \#error so that language implementers know what they need to provide an implementation for to use this feature.